torchtune/training/memory.py

```diff
@@ -304,3 +304,68 @@ def log_memory_stats(
         f"\n\t{device_support.device_name} peak memory reserved: {stats['peak_memory_reserved']:.2f} GiB"
         f"\n\t{device_support.device_name} peak memory active: {stats['peak_memory_active']:.2f} GiB"
     )
+
+
+class Float8Handler:
+    def __init__(self, enable_fp8_training: bool, dp_shard: int):
+        self.enabled = enable_fp8_training
+
+        if not self.enabled:
+            return
+        try:
+            from torchao.float8 import Float8LinearConfig
+            from torchao.float8.config import CastConfig, ScalingType
+        except ImportError as e:
+            raise ImportError(
+                "torchao is not installed. Please install it to use float8 linear layers."
+            ) from e
+
+        dp_shard_enabled = dp_shard > 1
+        # Mutates the model inplace replacing instances of torch.nn.Linear with Float8Linear
+
+        # in torchtitan they also check if this is explicitly set - this implementation is
+        # more opinionated if fsdp
+        enable_fsdp_float8_all_gather = dp_shard_enabled
+        self.precompute_scale = enable_fsdp_float8_all_gather
```

Comment:
nit: can probably just do this?
```suggestion
        self.precompute_scale = dp_shard_enabled
```